libmozevent/storage.py

```diff
@@ -12,37 +10,23 @@
 
 
 class EphemeralStorage:
-    @classmethod
-    async def create(cls, name, expiration):
-        self = EphemeralStorage()
+    def __init__(self, name, expiration):
         self.name = name
         self.expiration = expiration
         self.cache = {}
         self.redis_enabled = "REDIS_URL" in os.environ
         logger.info("Redis support", enabled=self.redis_enabled and "yes" or "no")
 
-        if self.redis_enabled:
-            async with AsyncRedis() as redis:
-                # Remove all expired keys from our sorted set.
-                await redis.zremrangebyscore(
-                    self.name, min=-math.inf, max=int(time.time())
-                )
-                # Load all live keys from our sorted set.
-                keys = await redis.zrangebyscore(
-                    self.name, min=int(time.time()), max=math.inf
-                )
-                for key in keys:
-                    key = key.decode("utf-8")
-                    self.cache[key] = pickle.loads(
-                        await redis.get(f"{self.name}:{key}")
-                    )
-
-        return self
+    def _redis_key(self, key):
+        return f"{self.name}:{key}"
 
-    def __len__(self):
-        return len(self.cache)
+    async def get(self, key):
+        if self.redis_enabled and key not in self.cache:
+            async with AsyncRedis() as redis:
+                val = await redis.get(self._redis_key(key))
+                if val is not None:
+                    self.cache[key] = pickle.loads(val)
```

Comment:
No protocol ?

Reply:
loads detects it automatically (from https://docs.python.org/3/library/pickle.html#pickle.loads:
> The protocol version of the pickle is detected automatically, so no protocol argument is needed. Bytes past the pickled representation of the object are ignored.

I could define pickle.HIGHEST_PROTOCOL for dumps, but better to change that in a separate PR for both this module and the bus module.

Reply:
I've filed https://github.com/mozilla/libmozevent/issues/51.